src/services/transactionService.js

```diff
@@ -0,0 +1,42 @@
+import localforage from 'localforage';
+import walletService from './walletService';
+import { BASE_CURRENCY, TRANSACTIONS_HISTORY_KEY } from '../utils/constants';
+
+export const isUserBuying = (amount) => amount > 0;
+
+const isCurrencySufficient = async ({ userId, currencyName, amount, price }, shouldCheckBaseCurrency) => {
+  const availableFunds = await walletService.getAvailableUserFunds(
+    userId,
+    shouldCheckBaseCurrency ? BASE_CURRENCY : currencyName
+  );
+  const amountToPay = shouldCheckBaseCurrency ? price * amount : Math.abs(amount);
+  return availableFunds >= amountToPay;
+};
+
+export const isBaseCurrencySufficient = async (transactionInfo) => isCurrencySufficient(transactionInfo, true);
```

Comment:
I'd suggest to avoid that `true/false` arguments. Without implementation inspection developer won't know what is this flag related to.